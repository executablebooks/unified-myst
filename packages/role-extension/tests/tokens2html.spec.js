import { micromark } from 'micromark'

import { mystRoleMmarkExt, mystRoleHtmlExt } from '../src/index.js'

const fixtures = [
    '{}',
    '{ }',
    '{name}',
    '{name}`',
    '{name} `content`',
    '{}`content`',
    '{ }`content`',
    '{ a}`content`',
    '{name}`content`',
]

describe('Convert source to HTML', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystRoleMmarkExt],
            htmlExtensions: [mystRoleHtmlExt],
        }
        const htmlText = micromark(content, options)
        expect(htmlText).toMatchSnapshot()
    })
})
