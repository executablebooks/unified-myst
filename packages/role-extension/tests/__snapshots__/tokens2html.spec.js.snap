// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to HTML "{ }" 1`] = `"<p>{ }</p>"`;

exports[`Convert source to HTML "{ }\`content\`" 1`] = `"<p>{ }<code>content</code></p>"`;

exports[`Convert source to HTML "{ a}\`content\`" 1`] = `"<p>{ a}<code>content</code></p>"`;

exports[`Convert source to HTML "{}" 1`] = `"<p>{}</p>"`;

exports[`Convert source to HTML "{}\`content\`" 1`] = `"<p>{}<code>content</code></p>"`;

exports[`Convert source to HTML "{name} \`content\`" 1`] = `"<p>{name} <code>content</code></p>"`;

exports[`Convert source to HTML "{name}" 1`] = `"<p>{name}</p>"`;

exports[`Convert source to HTML "{name}\`" 1`] = `"<p>{name}\`</p>"`;

exports[`Convert source to HTML "{name}\`content\`" 1`] = `"<p><span class=\\"myst-role\\">name<code>content</code></span></p>"`;
